### DISCUSSION TITLE
TailwindCSS group-hover not working with dynamic values (Nuxt 3)

### DISCUSSION THREAD

**Discussion by OmicronZed** [tailwindlabs/tailwindcss#13040] [Help] [answered]
Hey everyone,

i have been trying to use "group-hover" with dynamic color codes for several days. If I assign a static value (`:class="group-hover:text-red-900"`) then it works.

But my color codes are fetched with `useAsyncData`. They do not work there.

```javascript
const data = await useAsyncData("collections", () =>
  $fetch("/api/collections")
);
```

This is an example of what data is:
```javascript
[ {
    title: "example1",
    handle: "example1",
    image: "https://example.com/example1",
    color: "#ffc700",
},
{
    title: "example2",
    handle: "example2",
    image: "https://example.com/example2",
    color: "#ffc700",
},
{
    title: "example3",
    handle: "example3",
    image: "https://example.com/example3",
    color: "#ffc700",
}]
```

And here is the template:
```vue
<template>
  <div class="bg-white w-full h-12 flex justify-center">
    <div class="container bg-black/0 h-full flex w-full justify-between">
        <template v-for="i in data">
            <div class="group flex h-full items-center cursor-pointer select-none">
                <p :class="[`group-hover:text-[${i.color}]`]">{{ i.title }}</p>
                <NuxtImg class="h-4 ml-2" :src="i.image" />
            </div>
        </template>
    </div>
  </div>
</template>
```

In the source code of the HTML document it is displayed correctly.
![image](https://github.com/tailwindlabs/tailwindcss/assets/44533440/6e685afa-dc56-4d10-a958-29f71e6bf2af)

Could it be that there is a problem with the render order?

I am referring to the following documentation: [TailwindCSS](https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state)

I am using Nuxt 3.10.3 and TailwindCSS 3.4.1.

Thanks in advance

**Comment by alex-krasikau** [comment, marked as answer]
Hi @OmicronZed,

Let's simplify this. Tailwind CSS operates directly on your source code. Unlike some tools, it doesn't examine the HTML output after it's been rendered. Instead, it reads through your source code during the build process and generates CSS classes based on what it finds.

If you want a design element on your page to change dynamically, a common method is using CSS variables. Although I'm primarily not a Vue developer, the outline of the solution, using your parameters, could look something like this:

```<p :style="`--text-color: ${i.color}`" class="group-hover:text-[--text-color]">{{ i.title }}</p>```

This line of code essentially binds the CSS variable '--text-color' to 'i.color' using the Vue 'style' directive. If the group is hovered over, the text color will change to whatever color 'i.color' is currently set to, and the title of 'i' will be displayed.

**Reply by OmicronZed** [reply]
Thank you! I have been trying to solve the problem for days! <3
